File: python/2.py
""" File: 2.py
    Date: 03/01/2010
    Name: Salvador Fuentes Jr.
    Info: Euler Project Problem/Exercise #1 
"""

def main():
    sum_ubound = 4000000
    i = 2
    fibSeq = [1,2]
    curr_sum = 3 

    while True:
        newFibNum = fibSeq[i-1] + fibSeq[i-2]
        if curr_sum + newFibNum > sum_ubound:
            break

        i += 1
        fibSeq.append(newFibNum)
        curr_sum += newFibNum

    print curr_sum     
    print fibSeq     

if __name__ == "__main__":
    main()

File: haskell/2.hs
{-  File: 2.hs
    Date: 02/28/2010
    Name: Salvador Fuentes Jr.
    Info: Euler Project Problem/Exercise #2
-}

module Main ( main ) where

main :: IO ()
main = print solveEuler2

solveEuler2 :: Int
solveEuler2 = last (takeWhile (<5000000) (scanl (+) 0 evenFibs))

evenFibs = filter even fibSeq 

fibSeq :: [Int]
fibSeq = 1 : 2 : zipWith (+) fibSeq (tail fibSeq)


{- Fib function 
                    

fib :: Int -> Int
fib x = fib (x - 1) + fib (x - 2)

fib x   | x > 2 = (fib x-1) + (fib x-2) 
        | otherwise = x

-}
